apps/rush-lib/assets/rush-init/rush.json

```diff
@@ -61,7 +61,21 @@
      * After modifying this field, it's recommended to run "rush update --full" so that the package manager
      * will recalculate all version selections.
      */
-    /*[LINE "HYPOTHETICAL"]*/ "resolutionStrategy": "fast"
+    /*[LINE "HYPOTHETICAL"]*/ "resolutionStrategy": "fast",
+
+    /**
+     * This defines the granularity by which external dependencies are tracked to detect if a project needs
+     * to be rebuilt during an incremental build.
+     *
+     * If set to "project," Rush will track only the individual project's direct and indirect dependencies when
+     * detecting if a project needs to be rebuilt.
+     *
+     * If set to "repository," all projects will be rebuilt if any external dependencies anywhere in the repository
+     * are changed.
+     *
```

Comment:
I like it. Minor suggestion. 
How about:

     * If set to "repository," all projects will be rebuilt if any external dependencies anywhere in the repository
     * have changed.

Or maybe:
 
     * If set to "repository," all projects will be rebuilt if any external dependency anywhere in the repository
     * has changed.

Reply:
How would a repo maintainer decide whether to specify `"project"` versus `"repository"`?

Reply:
I chatted with @iclanton about this.  We realized that `incrementalBuildDependencyGranularity`  is not a proper setting:  The `project` mode is the correct implementation, and the `repository` mode is the worse old implementation.  Nobody would choose `repository` except to temporarily revert to the old behavior until a fix is published.

Thus instead of adding this to `rush.json`, we're thinking to introduce an optional config file that we can use to opt-in or opt-out of new features.  I'm thinking it should be restricted to boolean flags, like a flight or kill switch.  Something like this:

**common/config/rush/experiments.json**
```ts
{
  /**
   * Rush 5.14.0 improved incremental builds to ignore spurious changes in the pnpm-lock.json file.
   * This optimization is enabled by default.  If you encounter a problem where "rush build" is neglecting
   * to build some projects, please open a GitHub issue.  As a workaround you can uncomment this line
   * to temporarily restore the old behavior where everything must be rebuilt whenever pnpm-lock.json
   * is modified.
   */
  // "legacyIncrementalBuildDetection": true
}
```